### PR TITLE
Generate minimal OWL file using ASCT+B terms

### LINKED ISSUE #111: generate minimal OWL file using ASCT+B terms as seed
* Step 1:  Using all AS & CT terms from a single table, use SPARQL to query the *redundant* graph for all edges connecting terms in the seed, following the same limited set of properties we use for constructing AS_CT.owl.   Use SPARQL construct to generate a existential restrictions for all of these. (Not sure how well this will scale for a single SPARQL query)
* Step2:  Use Sparql to check whether properties use are transitive & add this with SPARQL construct  
* Step3: Use ROBOT reduce to save the results to an OWL output file
* Step4: Generate an graph to visualise

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #111

### CODE REVIEW

File: src/template_generation_tools.py [hunk 1]
```diff
@@ -100,6 +100,8 @@ def generate_class_graph_template(ccf_tools_df :pd.DataFrame):
 
   terms_ct_as_start = len(terms_ct_as)    
 
+  all_edges = ug.construct_uberon(" ".join(list(terms_pairs.union(terms_ct_as))), ug.construct_all_edges)
```
Comment (dosumis) [inline src/template_generation_tools.py:103]: Is this sufficient to generate all possible combinations of edges for testing in the where clause of construct_all_edges?  I'm wondering if this is why some edges are being dropped - e.g. in https://github.com/obophenotype/CCF_tools/issues/111#issuecomment-980091830


1. Some terms in the first graph are singletons but should be wired up, e.g.  
image
should have a part_of relationship to 'kidney interstitum' - these terms are missing from graph 2
3. Some other edges missing, e.g. 'renal efferent arteriole' should be under 'kidney vasculature' and under 'renal corpuscle'

**Review by dosumis — CHANGES_REQUESTED** [review]
See question

File: src/template_generation_tools.py (hunk 1 above)
Comment (dosumis) [reply src/template_generation_tools.py:103]: Potential fix:
1. SELECT all triples from redundant graph with part_of in subject position and one of allowed OPs as predicate = set S
1. SELECT all triples from redundant graph with part_of in object position and one of allowed OPs as predicate = set  O
2. Generate intersection of sets S & O and use these to generate triples with SPARQL construct (as now)

**Comment by dosumis** [comment]
Could you push the changes?  I'd like to have a look at the OWL file.

**Comment by diane-psc** [comment]
Can this be merged? If not, can we archive it?

**Pull request closed without merging**